lib/methods/ifwapichecktask.cpp

```diff
@@ -0,0 +1,400 @@
+/* Icinga 2 | (c) 2023 Icinga GmbH | GPLv2+ */
+
+#ifndef _WIN32
+#	include <stdlib.h>
+#endif /* _WIN32 */
+#include "methods/ifwapichecktask.hpp"
+#include "icinga/icingaapplication.hpp"
+#include "icinga/pluginutility.hpp"
+#include "base/base64.hpp"
+#include "base/defer.hpp"
+#include "base/utility.hpp"
+#include "base/perfdatavalue.hpp"
+#include "base/convert.hpp"
+#include "base/function.hpp"
+#include "base/io-engine.hpp"
+#include "base/json.hpp"
+#include "base/logger.hpp"
+#include "base/shared.hpp"
+#include "base/tcpsocket.hpp"
+#include "base/tlsstream.hpp"
+#include "remote/apilistener.hpp"
+#include "remote/url.hpp"
+#include <boost/asio.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <exception>
+
+using namespace icinga;
+
+REGISTER_FUNCTION_NONCONST(Internal, IfwApiCheck, &IfwApiCheckTask::ScriptFunc, "checkable:cr:resolvedMacros:useResolvedMacros");
+
+static void ReportIfwCheckResult(
+	const Checkable::Ptr& checkable, const CheckCommand::Ptr& command, const CheckResult::Ptr& cr, const String& output,
+	double start, double end, int exitcode = 3, const Array::Ptr& perfdata = nullptr
+)
+{
+	if (Checkable::ExecuteCommandProcessFinishedHandler) {
+		ProcessResult pr;
+		pr.PID = -1;
+		pr.Output = perfdata ? output + " |" + String(perfdata->Join(" ")) : output;
+		pr.ExecutionStart = start;
+		pr.ExecutionEnd = end;
+		pr.ExitStatus = exitcode;
+
+		Checkable::ExecuteCommandProcessFinishedHandler(command->GetName(), pr);
+	} else {
+		cr->SetOutput(output);
+		cr->SetPerformanceData(perfdata);
+		cr->SetState((ServiceState)exitcode);
+		cr->SetExitStatus(exitcode);
+		cr->SetExecutionStart(start);
+		cr->SetExecutionEnd(end);
+		cr->SetCommand(command->GetName());
+
+		checkable->ProcessCheckResult(cr);
+	}
+}
```

Comment:
> show the URL instead of the command name (which is included in the URL), but that could be too much clutter in the end.

Well, we want to help the user with debugging...

```
$ curl -fsSku root:123456 -X GET -H 'Accept: application/json' 'https://127.0.0.1:5665/v1/objects/services/ifw-api!invalid-expected-sct?pretty=1&attrs=last_check_result'
{
    "results": [
        {
            "attrs": {
                "last_check_result": {
                    "active": true,
                    "check_source": "alexandersmbp2.int.netways.de",
                    "command": [
                        "curl",
                        "--silent",
                        "--show-error",
                        "--tlsv1.2",
                        "--fail",
                        "--resolve",
                        "invalid-expected-sct.badssl.com:443:invalid-expected-sct.badssl.com",
                        "--ciphers",
                        "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:AES256-GCM-SHA384:AES128-GCM-SHA256",
                        "--cert",
                        "/Users/aklimov/NET/WS/icinga2/prefix/var/lib/icinga2/certs//alexandersmbp2.int.netways.de.crt",
                        "--key",
                        "/Users/aklimov/NET/WS/icinga2/prefix/var/lib/icinga2/certs//alexandersmbp2.int.netways.de.key",
                        "--cacert",
                        "/Users/aklimov/Downloads/ISRG Root X1.pem",
                        "--request",
                        "POST",
                        "--url",
                        "https://invalid-expected-sct.badssl.com:443/v1/checker?command=ifw-api",
                        "--header",
                        "Content-Type: application/json",
                        "--data-raw",
                        "{}"
                    ],
                    "execution_end": 1685963018.725296,
                    "execution_start": 1685963018.370604,
                    "exit_status": 3,
                    "output": "Certificate validation failed for IfW API on host 'invalid-expected-sct.badssl.com' (SNI: 'invalid-expected-sct.badssl.com'; CN: 'invalid-expected-sct.badssl.com') port '443': code 10: certificate has expired",
                    "performance_data": null,
                    "previous_hard_state": 99,
                    "schedule_end": 1685963018.725378,
                    "schedule_start": 1685963018.364384,
                    "scheduling_source": "alexandersmbp2.int.netways.de",
                    "state": 3,
                    "ttl": 0,
                    "type": "CheckResult",
                    "vars_after": {
                        "attempt": 2,
                        "reachable": true,
                        "state": 3,
                        "state_type": 0
                    },
                    "vars_before": null
                }
            },
            "joins": {},
            "meta": {},
            "name": "ifw-api!invalid-expected-sct",
            "type": "Service"
        }
    ]
}
$
```

Reply:
I really don't understand, from "the URL [...] could be too much clutter", how does one derive that obviously, a full curl command line is the answer? I mean not suggesting it but implementing it right away.

Reply:
The only purpose of the gap you asked me to fill is helping the user debugging. This new command can -and therefore should- make it for the user as easy as possible.